tests/test_openshift.py

```diff
@@ -0,0 +1,478 @@
+# -*- coding: utf-8 -*-
+"""Unit tests for Openshift client."""
+import os
+from random import choice
+
+import pytest
+import mock
+import fauxfactory
+from wait_for import wait_for
+
+from wrapanapi.containers.providers import openshift
+from wrapanapi.containers.project import Project
+from wrapanapi.containers.deployment_config import DeploymentConfig
+from wrapanapi.containers.image_registry import ImageRegistry
+from wrapanapi.containers.image import Image
+from wrapanapi.containers.pod import Pod
+from wrapanapi.containers.service import Service
+from wrapanapi.containers.node import Node
+from wrapanapi.containers.replicator import Replicator
+from wrapanapi.containers.route import Route
+from wrapanapi.containers.template import Template
+from wrapanapi.containers.volume import Volume
+
+from wrapanapi.exceptions import InvalidValueException, RequestFailedException
+
+
+# Specify whether to use a mock provider or real one.
+MOCKED = os.environ.get('MOCKED', 'true').lower() == 'true'
```

Comment:
just for documentation we might want to consider putting the mocked/non-mocked variants into plugin objects at some point

while doing that we could also create a better handling than the env vars provide